bench-tps/src/bench.rs

```diff
@@ -154,7 +154,7 @@ struct TransactionChunkGenerator<'a, 'b, T: ?Sized> {
 
 impl<'a, 'b, T> TransactionChunkGenerator<'a, 'b, T>
 where
-    T: 'static + BenchTpsClient + Send + Sync + ?Sized,
+    T: 'static + GenericClient + Send + Sync + ?Sized,
```

Comment:
I think most of the changes is caused by the refactoring BenchTpsClient to GenericClient. I wonder why do we use BenchTpsClient not TpuClient directly -- what is the additional functionality we are getting? This background info can be added in the PR description to guide reviewers. In addition, we already have a client crate, I am concerned this generic-client make the client wrappings more complicated about which crate to use. Just some initial questions

Reply:
Done

> Also moves BenchTPSClient out of the bench-tps crate and into its own crate and renames it GenericClient, as it's now used by multiple crates (even before this change). Also adds functions to the GenericClient trait that solana-ping needs

Reply:
I preferred to do this just because bench-tps-client has most of the functionality needed by solana-ping, so it made the most sense to reuse that code. On the other hand, since BenchTPSClient is no longer being used by just bench-tps, I thought it was appropriate to rename it and move it into its own module.